Makefile

```diff
@@ -1,71 +1,87 @@
-.PHONY: generate
-generate:
-generate: generate/migrations
-generate: generate/sqlc
+.DEFAULT_GOAL := help
 
 .PHONY: db/reset
-db/reset: ## drop, create, and migrate dev and test databases
+db/reset: ## Drop, create, and migrate dev and test databases
 db/reset: db/reset/dev
 db/reset: db/reset/test
 
 .PHONY: db/reset/dev
-db/reset/dev: ## drop, create, and migrate dev database
+db/reset/dev: ## Drop, create, and migrate dev database
 	dropdb river_dev --force --if-exists
 	createdb river_dev
 	cd cmd/river && go run . migrate-up --database-url "postgres://localhost/river_dev"
 
 .PHONY: db/reset/test
-db/reset/test: ## drop, create, and migrate test databases
+db/reset/test: ## Drop, create, and migrate test databases
 	go run ./internal/cmd/testdbman reset
 
+.PHONY: generate
+generate: ## Generate generated artifacts
+generate: generate/migrations
+generate: generate/sqlc
+
 .PHONY: generate/migrations
-generate/migrations: ## sync changes of pgxv5 migrations to database/sql
+generate/migrations: ## Sync changes of pgxv5 migrations to database/sql
 	rsync -au --delete "riverdriver/riverpgxv5/migration/" "riverdriver/riverdatabasesql/migration/"
 
 .PHONY: generate/sqlc
-generate/sqlc:
+generate/sqlc: ## Generate sqlc
 	cd riverdriver/riverdatabasesql/internal/dbsqlc && sqlc generate
 	cd riverdriver/riverpgxv5/internal/dbsqlc && sqlc generate
 
+# Looks at comments using ## on targets and uses them to produce a help output.
+.PHONY: help
+help: ALIGN=22
+help: ## Print this message
+	@awk -F '::? .*## ' -- "/^[^':]+::? .*## /"' { printf "'$$(tput bold)'%-$(ALIGN)s'$$(tput sgr0)' %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+
+# Each directory of a submodule in the Go workspace. Go commands provide no
+# built-in way to run for all workspace submodules. Add a new submodule to the
+# workspace with `go work use ./driver/new`.
+submodules := $(shell go list -f '{{.Dir}}' -m)
+
+# Definitions of following tasks look ugly, but they're done this way because to
+# produce the best/most comprehensible output by far (e.g. compared to a shell
+# loop).
 .PHONY: lint
-lint:
-	cd . && golangci-lint run --fix
-	cd cmd/river && golangci-lint run --fix
-	cd riverdriver && golangci-lint run --fix
-	cd riverdriver/riverdatabasesql && golangci-lint run --fix
-	cd riverdriver/riverpgxv5 && golangci-lint run --fix
-	cd rivershared && golangci-lint run --fix
-	cd rivertype && golangci-lint run --fix
+lint:: ## Run linter (golangci-lint) for all submodules
+define lint-target
+    lint:: ; cd $1 && golangci-lint run --fix
+endef
+$(foreach mod,$(submodules),$(eval $(call lint-target,$(mod))))
```

Comment:
This looks a little weird, but out of all the options I tried, it produces by far the best output. Basically equivalent to what we had before:

```
$ make lint
cd /Users/brandur/Documents/projects/river && golangci-lint run --fix
cd /Users/brandur/Documents/projects/river/cmd/river && golangci-lint run --fix
cd /Users/brandur/Documents/projects/river/riverdriver && golangci-lint run --fix
cd /Users/brandur/Documents/projects/river/riverdriver/riverdatabasesql && golangci-lint run --fix
cd /Users/brandur/Documents/projects/river/riverdriver/riverpgxv5 && golangci-lint run --fix
cd /Users/brandur/Documents/projects/river/rivershared && golangci-lint run --fix
cd /Users/brandur/Documents/projects/river/rivertype && golangci-lint run --fix
```

The problem with using something like a `for` loop is that you just get the for loop printed, so the output is completely useless because you can't tell which module is currently running.